setup/package2.py

```diff
@@ -0,0 +1,569 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import argparse
+import logging
+import os
+import pexpect
+import shutil
+import signal
+import subprocess
+import sys
+import tempfile
+import time
+import traceback
+try:
+    from xmlrpc import client as xmlrpclib
+except ImportError:
+    import xmlrpclib
+
+from glob import glob
+
+#----------------------------------------------------------
+# Utils
+#----------------------------------------------------------
+
+ROOTDIR = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+TSTAMP = time.strftime("%Y%m%d", time.gmtime())
+exec(open(os.path.join(ROOTDIR, 'odoo', 'release.py'), 'rb').read())
+VERSION = version.split('-')[0].replace('saas~', '')
+GPGPASSPHRASE = os.getenv('GPGPASSPHRASE')
+GPGID = os.getenv('GPGID')
+DOCKERVERSION = VERSION.replace('+', '')
+INSTALL_TIMEOUT = 600
+
+DOCKERUSER = """
+RUN mkdir /var/lib/odoo && \
+    groupadd -g %(group_id)s odoo && \
+    useradd -u %(user_id)s -g odoo odoo -d /var/lib/odoo && \
+    mkdir /data && \
+    chown odoo:odoo /var/lib/odoo /data
+USER odoo
+""" % {'group_id': os.getgid(), 'user_id': os.getuid()}
+
+
+def run_cmd(cmd, chdir=None, timeout=None):
+    logging.info("Running command %s", cmd)
+    return subprocess.run(cmd, cwd=chdir, timeout=timeout)
+
+
+def _rpc_count_modules(addr='http://127.0.0.1', port=8069, dbname='mycompany'):
+    time.sleep(5)
+    uid = xmlrpclib.ServerProxy('%s:%s/xmlrpc/common' % (addr, port)).authenticate(
+        dbname, 'admin', 'admin', {}
+    )
+    modules = xmlrpclib.ServerProxy('%s:%s/xmlrpc/object' % (addr, port)).execute(
+        dbname, uid, 'admin', 'ir.module.module', 'search', [('state', '=', 'installed')]
+    )
+    if modules and len(modules) > 1:
+        time.sleep(1)
+        toinstallmodules = xmlrpclib.ServerProxy('%s:%s/xmlrpc/object' % (addr, port)).execute(
+            dbname, uid, 'admin', 'ir.module.module', 'search', [('state', '=', 'to install')]
+        )
+        if toinstallmodules:
+            logging.error("Package test: FAILED. Not able to install dependencies of base.")
+            raise Exception("Installation of package failed")
```

Comment:
Is the exception message hard-coded? Why not raise the exception with the logging message thing? Or is the exception also swallowed by something else?